streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java

```diff
@@ -813,6 +825,137 @@ public void shouldReadFutureProcessFileFormat() throws Exception {
         assertThat(directory.initializeProcessId(), equalTo(processId));
     }
 
+    @Test
+    public void shouldNotInitializeStandbyTasksWhenNoLocalState() {
+        initializeTasksForLocalState(new TaskId(0, 0), false);
+        assertFalse(directory.hasPendingTasks());
+    }
+
+    @Test
+    public void shouldInitializeStandbyTasksForLocalState() {
+        initializeTasksForLocalState(new TaskId(0, 0), true);
+        assertTrue(directory.hasPendingTasks());
+    }
+
+    @Test
+    public void shouldNotAssignPendingTasksWeDontHave() {
+        final TaskId taskId = new TaskId(0, 0);
+        initializeTasksForLocalState(taskId, false);
+        final Task task = directory.removePendingTask(taskId);
+        assertNull(task);
+    }
```

Comment:
How is this different from `shouldNotInitializeStandbyTasksWhenNoLocalState()`?

Reply:
The main difference here is that one is testing `directory.hasPendingTasks()` and the other tests `directory.removePendingTask`.

Reply:
Ahh yeah, these tests could definitely be better. I've adopted your suggestion here: 82e1f1e